expression/typeinfer_test.go

```diff
@@ -42,125 +42,17 @@ type typeInferTestCase struct {
 	decimal int
 }
 
-type testInferTypeSuite struct {
+type InferTypeSuite struct {
```

Comment:
I guess you can just inline the Parser instance where it is required.

Reply:
I have removed it.